plugins/plugin-react-refresh/plugin.js

```diff
@@ -45,25 +46,27 @@ async function transformJs(contents, id, skipTransform) {
     );
     fastRefreshEnhancedCode = contents;
   } else {
-    const {code} = await babel.transformAsync(contents, {
+    const {code, map} = await babel.transformAsync(contents, {
       cwd: process.cwd(),
       filename: id,
       ast: false,
       compact: false,
-      sourceMaps: false,
+      sourceMaps,
+      inputSourceMap: inputSourceMap ? JSON.parse(inputSourceMap) : undefined,
       configFile: false,
       babelrc: false,
       plugins: [require('react-refresh/babel'), require('@babel/plugin-syntax-class-properties')],
     });
     fastRefreshEnhancedCode = code;
+    outputMap = map;
   }
 
   // If fast refresh markup wasn't added, just return the original content.
   if (!fastRefreshEnhancedCode || !IS_FAST_REFRESH_ENABLED.test(fastRefreshEnhancedCode)) {
     return contents;
   }
 
-  return `
+  const code = `
 /** React Refresh: Setup **/
 if (import.meta.hot) {
```

Comment:
If you write a babel plugin to insert those codes, there is no need to handle source map changes.

Reply:
Thanks a lot.
[I have written a babel plugin to implement it. ](https://github.com/magic-akari/snowpack/blob/fix-sourcemap-dev-mode-react-refresh/plugins/plugin-react-refresh/hmr-babel-plugin.js)

However, there is still [somewhere](https://github.com/snowpackjs/snowpack/blob/3783376c1e19fc09839fd979141c753becd0c78a/snowpack/src/build/build-import-proxy.ts#L24-L54) in the main snowpack package where changed the code without correct sourcemap.
But there is not babel installed in snowpack's package.json, I am not sure whether babel should be imported.